### PR TITLE
Bump `kotlinStdLib`, `coroutinesAndroid` and JaCoCo dependency version

### PR DESCRIPTION
**Pull request by Guardiola31337**
### Description

Bumps `kotlinStdLib`, `coroutinesAndroid` and JaCoCo dependency version to [`1.5.0`](https://blog.jetbrains.com/kotlin/2021/05/kotlin-1-5-0-released/), `1.4.3` and `0.8.7` respectively

### CODE REVIEW

**Comment by Guardiola31337** [comment]
This is blocked by https://github.com/jacoco/jacoco/issues/1155

Reason why `unit-tests` are failing

```
Caused by: java.lang.IllegalStateException: Unexpected SMAP line: *S KotlinDebug
        at org.jacoco.core.internal.analysis.filter.KotlinInlineFilter.getFirstGeneratedLineNumber(KotlinInlineFilter.java:98)
        at org.jacoco.core.internal.analysis.filter.KotlinInlineFilter.filter(KotlinInlineFilter.java:44)
        at org.jacoco.core.internal.analysis.filter.Filters.filter(Filters.java:58)
        at org.jacoco.core.internal.analysis.ClassAnalyzer.addMethodCoverage(ClassAnalyzer.java:119)
        at org.jacoco.core.internal.analysis.ClassAnalyzer.access$100(ClassAnalyzer.java:33)
        at org.jacoco.core.internal.analysis.ClassAnalyzer$1.accept(ClassAnalyzer.java:108)
        at org.jacoco.core.internal.flow.ClassProbesAdapter$2.visitEnd(ClassProbesAdapter.java:91)
        at org.objectweb.asm.ClassReader.readMethod(ClassReader.java:1288)
        at org.objectweb.asm.ClassReader.accept(ClassReader.java:688)
        at org.objectweb.asm.ClassReader.accept(ClassReader.java:400)
        at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:116)
        at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:132)
        ... 208 more
```

Although a fix for it has landed, JaCoCo `0.8.7` hasn't been released yet. When available we should bump JaCoCo to `0.8.7` in https://github.com/mapbox/mapbox-navigation-android/blob/133d6e1ef20b1f9192f1024c64030cf7db5938b7/gradle/jacoco.gradle#L4 and that should fix the crash.

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -163,7 +163,7 @@ ext {
       gradle               : '4.0.1',
       dependencyGraph      : '0.5.0',
       dependencyUpdates    : '0.29.0',
-      kotlin               : '1.4.30',
+      kotlin               : '1.5.0',
```
Comment (RingerJK) [inline gradle/dependencies.gradle:166]: `kotlinStdLib` must be bumped as well.

Comment (Guardiola31337) [reply gradle/dependencies.gradle:166]: Great catch!

**Comment by Guardiola31337** [comment]
This is not blocked anymore, [JaCoCo `0.8.7`](https://github.com/jacoco/jacoco/releases/tag/v0.8.7) went out 🎉 

Ready for another round of 👀

**Review by LukasPaczos — APPROVED** [review]

**Pull request merged**